ehcache-impl/src/main/java/org/ehcache/impl/internal/store/tiering/TieredStore.java

```diff
@@ -572,7 +573,11 @@ public NoopCachingTier(final AuthoritativeTier<K, V> authoritativeTier) {
     @Override
     public ValueHolder<V> getOrComputeIfAbsent(final K key, final Function<K, ValueHolder<V>> source) {
       final ValueHolder<V> apply = source.apply(key);
-      authoritativeTier.flush(key, apply);
+      Optional<ValueHolder<V>> ap = Optional.ofNullable(apply);
```

Comment:
Why the optional? It seems like all this does is create an object for no real benefit?